Branch-SDK/src/io/branch/referral/Branch.java

```diff
@@ -2945,7 +2947,9 @@ private void registerInstallOrOpen(ServerRequest req, BranchReferralInitListener
         // be cleared if the app is terminated while an Open/Install is pending.
         else {
             // Update the callback to the latest one in initsession call
-            deferredInitListener_ = new WeakReference<>(callback);
+            if (callback != null) {
+                deferredInitListener_ = new WeakReference<>(callback);
+            }
             requestQueue_.setInstallOrOpenCallback(callback);
```

Comment:
Good point sahil , We already do `if (callback != null) {` check inside the method `setInstallOrOpenCallback`